server/plugin/utils_test.go

```diff
@@ -356,6 +356,81 @@ func TestGetPluginURL(t *testing.T) {
 	}
 }
 
+func TestSanitizeURLPath(t *testing.T) {
+	p := Plugin{}
+	for _, testCase := range []struct {
+		description  string
+		actualPath   string
+		expectedPath string
+	}{
+		{
+			description:  "SanitizeURLPath: valid",
+			actualPath:   "/dummy_org/dummy_project",
+			expectedPath: "/dummy_org/dummy_project",
+		},
+		{
+			description:  "SanitizeURLPath: valid with dots in project name",
+			actualPath:   "/dummy_org/dummy..1..2..3..project/dummy.project/dummy....project",
+			expectedPath: "/dummy_org/dummy..1..2..3..project/dummy.project/dummy....project",
+		},
+		{
+			description:  "SanitizeURLPath: invalid variation 1",
+			actualPath:   "/../dummy_org/dummy_project",
+			expectedPath: "/dummy_org/dummy_project",
+		},
+		{
+			description:  "SanitizeURLPath: invalid variation 2",
+			actualPath:   "../dummy_org/../dummy_project/..",
```

Comment:
Shouldn't the expected value of this be `dummy_org/dummy_project` instead of starting with a forward slash?

Regarding `SanitizeURLPath`, I think the logic should be:

1. unescapedPath (replace %2e with . and %2f with /)
2. if path starts with / , a simple `path.Clean(actualPath)` should be enough
3. if path does not start with /, doing `strings.TrimPrefix("/", path.Join("/", actualPath))` should work properly

Example inputs and outputs ( opinion on this @jupenur ):

```
/dummy_org/dummy_project -> /dummy_org/dummy_project
/dummy_org/dummy..1..2..3..project/dummy.project/dummy....project -> /dummy_org/dummy..1..2..3..project/dummy.project/dummy....project
/../dummy_org/dummy_project -> /dummy_org/dummy_project
../dummy_org/../dummy_project/.. -> ""
../../dummy_org/../../dummy_project -> dummy_project
/dummy_org/../dummy_project ->  /dummy_project
/dummy_org/%2e%2e/dummy_project -> /dummy_project
/dummy_org/.%2e/dummy_project -> /dummy_project
/dummy_org/%2e./dummy_project -> /dummy_project
/dummy_org/.%2e/%2e./dummy_project -> /dummy_project
/dummy_org/.%2e%2e./dummy_project/.. -> /dummy_org/....
%2e%2e/dummy_org/.%2e%2e./dummy_project/.%2e.%2e/ -> dummy_org/..../dummy_project/....
```

Reply:
> 2. if path starts with / , a simple `path.Clean(actualPath)` should be enough

No. The path here is the actual full REST API path with inputs already baked in. Just calling `Clean` on it doesn't fix anything -- it just normalizes the path, but Azure is already doing that on their end before routing, so there's no impact there.

The thing we're trying to prevent here (which apparently is not clear) is this:

1. There is an API call to `/some/path/with/${inputs}`
2. The user sets `inputs` to `../../other/path`
3. We issue an API call to `/some/path/with/../../other/path`
4. Azure treats the call as a call to `/some/other/path`

Calling `Clean` does not solve this, because with `Clean` the path `/some/path/with/../../other/path` just becomes `/some/other/path`. There is no "escaping the root" because we're already dealing with the completed path. The root is what it is, and Azure takes care of that, but the input should not change the API endpoint being called.

Reply:
Adding to what @jupenur mentioned instead of checking the entire URL we should check and sanitize individual inputs that are going to be part of the URL, this way we can allow having only those characters in such inputs which are supported by Azure DevOps APIs removing all non-allowed chars like `@, /`, etc and also removing suspicious path traversal patterns formed by `.` and `%2e` like `../`, `/..` `/../..`, `/%2e%2e`, etc

let me know if this seems good to you and we proceed with it @jupenur @m1lt0n